### PR TITLE
feat: add time-picker

### PR DESCRIPTION
**Pull request by colinlienard**
- Closes #1376

## 📝 Description

Add a TimePicker component 

## 🚀 New behavior

- Input with the value as a string
- Trigger to open a menu
- Choose an hour, a minute and a period (for specific locales) in the menu

## 💣 Is this a breaking change (Yes/No):

No

## 📝 Additional Information

- [x] Basic logic
- [x] Complete API
	- [x] Clearable
	- [x] Disabled
	- [x] Seconds
	- [x] Steps
	- [x] Min/max hours/minutes
- [x] Props
- [x] Handle locale
- [x] Accessibility
- [x] Examples
- [x] Docs
- [x] Tests

### CODE REVIEW

File: packages/machines/time-picker/src/time-picker.anatomy.ts [hunk 1]
```diff
@@ -5,12 +5,13 @@ export const anatomy = createAnatomy("time-picker").parts(
...
   "clearTrigger",
   "content",
+  "contentColumn",
   "control",
   "input",
   "hourCell",
   "minuteCell",
-  "AMPeriodTrigger",
-  "PMPeriodTrigger",
+  "amPeriodTrigger",
```
Comment (segunadebayo) [inline packages/machines/time-picker/src/time-picker.anatomy.ts:13]: Consider using just `periodTrigger` here. For the prop getters, you can call it `getPeriodTriggerProps({ type: "am|pm" })`

**Comment by colinlienard** [comment]
I don't think I've forgotten anything major, but I'm curious to see from your review whether I did things right or not.
Just not sure how the anatomy part in the doc is done, so I skipped it

File: packages/machines/time-picker/src/time-picker.types.ts [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+   */
+  max?: Time
+  /**
+   * The steps of each time unit.
+   */
+  steps?: { hour?: number; minute?: number; second?: number }
+  /**
+   * Whether to show the seconds.
+   */
+  withSeconds?: boolean
```
Comment (maastrich) [inline packages/machines/time-picker/src/time-picker.types.ts:100]: I would suggest using a control a bit more agnostic:
```suggestion
  precision?: "hours" | "minutes" | "seconds" | "miliseconds"
```
This way, you may try and add "milliseconds" entry and also be able to use timePicker with hours only

**Comment by colinlienard** [comment]
Hey @segunadebayo any chance to get a review soon?
Let me know if I can do more
Thanks!

**Comment by segunadebayo** [comment]
Hi @colinlienard,

Thanks for your patience. Will review this shortly.

**Comment by pierresisson** [comment]
> Hi @colinlienard,
> 
> Thanks for your patience. I will review this shortly.
> 
> > I might do a video review instead and share the link here.

Hi @segunadebayo, any news on this Pull Request ?

Thank you!

**Comment by segunadebayo** [comment]
@colinlienard  @pierresisson Took the time to refactor the machine and APIs overall. I figured we might not merge this soon enough if I had to send a review.

**Pull request merged**